packages/sdk/src/AbstractConfigManager.ts

```diff
@@ -199,6 +199,40 @@ export abstract class AbstractConfigManager {
         };
     }
 
+    public async promptForDeployDirectory(defaultServerPath: string): Promise<string> {
+        const result = await this.showMenu({
+            items: [{ label: defaultServerPath }, { label: "$(plus) Add New Deploy Directory" }],
+            placeholder: `Select a deploy directory for the Zowe SSH server`,
+        });
+
+        if (!result) return;
+
+        if (result === "$(plus) Add New Deploy Directory") {
+            const input = await this.showInputBox({
+                title: "Enter custom deploy directory for the Zowe SSH server",
+                placeHolder: `E.g. ${defaultServerPath}`,
+                validateInput: (input) => {
+                    const trimmed = input.trim();
+                    if (!trimmed) return "Path cannot be empty";
+                    if (!path.isAbsolute(trimmed)) return "Deploy Directory must be an absolute path";
+                    return null;
+                },
+            });
+
+            return input?.trim() || defaultServerPath;
+        }
+
+        if (!path.isAbsolute(result)) {
+            this.showMessage(
+                "Invalid Deploy Directory format. Ensure it matches the expected pattern.",
+                MESSAGE_TYPE.ERROR,
+            );
+            return;
+        }
+
+        return result;
```

Comment:
If the user chose a custom deploy location, we need to make it persistent.

Before returning the result, can we call `Utilities.getVsceConfig().set("serverInstallPath", ...` to save the server path in VS Code settings? If they kept the default location, then we can skip this step.